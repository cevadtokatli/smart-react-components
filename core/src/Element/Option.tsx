import styled from 'styled-components'
import applyStyledProps, { StyledProps, shouldForwardProp } from '../styled-props'

export default styled.option.withConfig({ shouldForwardProp })<StyledProps>(applyStyledProps)
