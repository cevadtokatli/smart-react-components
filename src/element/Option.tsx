import styled from "styled-components"
import {Props, attrs, css} from "./"

export default styled.option.attrs<Props>(attrs)<Props>(css)