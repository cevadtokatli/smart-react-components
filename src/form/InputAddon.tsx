import styled from "styled-components"
import ElementProps from "../element/Props"
import Div from "../element/Div"

interface Props extends ElementProps {
    type: string
    shape: string
    absolute: boolean
}

export default styled(Div)<Props>(({theme,type,shape,absolute}) => `
    overflow: hidden;
    flex: 0 0 auto;
    display: inline-flex;

    ${!absolute ? `
        ${type == "left"
        ? `
            border-top-left-radius: ${theme.src.form.inputRadius[shape]};
            border-bottom-left-radius: ${theme.src.form.inputRadius[shape]};
        ` : `
            border-top-right-radius: ${theme.src.form.inputRadius[shape]};
            border-bottom-right-radius: ${theme.src.form.inputRadius[shape]};
        `}
    ` : `
        position: absolute;
    `}
`)