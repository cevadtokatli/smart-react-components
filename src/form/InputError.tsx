import styled from "styled-components"
import Div from "../element/Div"
import Props from "../element/Props"

export default styled(Div).attrs<Props>(({theme}) => ({
    mt: "2",
    color: theme.src.type.danger.main
}))``