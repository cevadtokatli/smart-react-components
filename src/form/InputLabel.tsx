import styled from "styled-components"
import ElementProps from "../element/Props"
import Label from "../element/Label"

interface Props extends ElementProps {
    disabled: boolean
}

export default styled(Label).attrs<Props>(({mb="2"}) => ({mb}))<Props>(({theme,disabled}) => `
    display: inline-block;
    word-wrap: break-word;
    word-break: break-word;

    ${disabled ? `
        opacity: ${theme.src.form.disabledOpacity};
        user-select: none;
    ` : ""}
`)