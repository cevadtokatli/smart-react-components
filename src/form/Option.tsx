import React from "react"
import {SizeProps} from "../props/size-props"
import {JSXElementProps, JSXChild, Value, ValueMix, SetValue} from "../props"
import DV from "../default-value"
import OptionElement from "./OptionElement"

export interface Props extends SizeProps {
    elementProps?: JSXElementProps
    children: JSXChild
    value: Value
    searchValue?: string
    disabled?: boolean
    box?: boolean
    type?: string
    fill?: boolean
    shape?: string
    hover?: boolean
    space?: boolean
    active?: ValueMix
    setActive?: SetValue<ValueMix>
}

const Option: React.FC<Props> = ({size="default",sizeSm,sizeMd,sizeLg,sizeXl,elementProps=DV.JSX_ELEMENT_PROPS,children,value,disabled,box,type="primary",fill,shape="default",hover,active,setActive}) => {
    /**
     * Returns active$ status of the element. 
     */
    const getActive$ = (): boolean => {
        if(value == null)
            return false
        return Array.isArray(active) ? active.indexOf(value) > -1 : value == active
    }

    const [active$, setActive$] = React.useState<boolean>(() => getActive$())
    
    React.useEffect(() => setActive$(getActive$()), [value, active])

    /**
     * Sets value if element is not disabled.
     */
    const set = (): void => {
        if(disabled || (value == null && Array.isArray(active)))
            return

        if(Array.isArray(active)) {
            if(active.indexOf(value) == -1)
                setActive([...active,value])
            else
                setActive(active.filter(i => i != value))
        } else if(value != active)
            setActive(value)
    }

    return <OptionElement {...elementProps} className={`${active$ ? "src-option-active" : ""}`} size={size} sizeSm={sizeSm} sizeMd={sizeMd} sizeLg={sizeLg} sizeXl={sizeXl} box={box} type={type} fill$={fill} shape={shape} hover={hover} disabled={disabled} value={value} onClick={set}>{children}</OptionElement>
}

export default Option