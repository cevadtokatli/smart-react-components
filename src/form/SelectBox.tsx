import React from "react"
import SelectProps from "./SelectProps"
import {JSXElementProps, JSXChild} from "../props"
import DV from "../default-value"
import useSelectMethods from "./UseSelectMethods"
import FixedBox, {FixedBoxOnRef} from "../fixed-box/FixedBox"
import useInputOutput from "./UseInputOutput"
import SelectElement from "./InputElement"
import SelectFixedBoxElement from "./SelectFixedBoxElement"
import Div from "../element/Div"
import Input from "./Input"

export interface Props extends SelectProps {
    boxProps?: JSXElementProps
    placeholder?: JSXChild
    badgeType?: string
    arrow?: boolean
    minWidth?: number
    boxShadow?: boolean
    transitionClassName?: string
    transitionType?: string
    transitionDuration?: number
    showAnimation?: boolean
    hideAnimation?: boolean
}

const SelectBox: React.FC<Props> = ({size="default",sizeSm,sizeMd,sizeLg,sizeXl,containerProps=DV.JSX_ELEMENT_PROPS,selectProps=DV.JSX_ELEMENT_PROPS,boxProps=DV.JSX_ELEMENT_PROPS,labelProps=DV.JSX_ELEMENT_PROPS,label,children,placeholder,optionType="primary",badgeType="primary",arrow=true,active,setActive,type="gray",fill,border,shape="default",hover=true,waveEffect="light",error,leftAddon,leftAddonProps=DV.JSX_ELEMENT_PROPS,leftAddonAbsolute,rightAddon,rightAddonProps=DV.JSX_ELEMENT_PROPS,rightAddonAbsolute,minWidth=300,boxShadow,transitionClassName,transitionType,transitionDuration,showAnimation=true,hideAnimation=true,search,searchInputProps=DV.JSX_ELEMENT_PROPS}) => {
    const {itemList,searchText,setSearchText} = useSelectMethods({size,sizeSm,sizeMd,sizeLg,sizeXl,box:true,placeholder,children,optionType,badgeType,arrow,fill,shape,hover,waveEffect,disabled:selectProps.disabled,active,setActive,closeBox:(() => setStatus(false)),search})
    const [status, setStatus] = React.useState<boolean>(false)
    const fixedBoxOnRef = React.useRef<FixedBoxOnRef>(null)

    React.useEffect(() => {
        if(fixedBoxOnRef.current)
            fixedBoxOnRef.current.setPosition()
    }, [itemList])

    React.useEffect(() => {
        if(!Array.isArray(active))
            setStatus(false)
    }, [active])

    return useInputOutput({
        element: (
            <FixedBox status={status} setStatus={setStatus} minWidth={minWidth} boxShadow={boxShadow} transitionClassName={transitionClassName} transitionType={transitionType} transitionDuration={transitionDuration} showAnimation={showAnimation} hideAnimation={hideAnimation} onRef={args => fixedBoxOnRef.current = args}>
                <SelectElement tabIndex={0} {...selectProps} size$={size} size$Sm={sizeSm} size$Md={sizeMd} size$Lg={sizeLg} size$Xl={sizeXl} type$={type} fill$={fill} border$={border} shape={shape} leftAddon={leftAddon ? true : false} leftAddonAbsolute={leftAddonAbsolute} rightAddon={rightAddon ? true : false} rightAddonAbsolute={rightAddonAbsolute} inputElement={false} data-src-placeholder-style={itemList.isPlaceholder} data-src-active={status}>{itemList.active}</SelectElement>
                <SelectFixedBoxElement size={size} sizeSm={sizeSm} sizeMd={sizeMd} sizeLg={sizeLg} sizeXl={sizeXl} {...boxProps} type={type} shape={shape}>
                    <Div height="inherit" minHeight="inherit" maxHeight="inherit">
                        {search && <section><Input {...searchInputProps} value={searchText} setValue={setSearchText} /></section>}
                        {itemList.value}
                    </Div>
                </SelectFixedBoxElement>
            </FixedBox>
        ),
        containerProps,inputProps:selectProps,labelProps,label,shape,leftAddon,leftAddonProps,leftAddonAbsolute,rightAddon,rightAddonProps,rightAddonAbsolute,error
    })
}

export default SelectBox