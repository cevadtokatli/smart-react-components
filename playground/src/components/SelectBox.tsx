import SelectBox from "@smart-react-components/ui/Select/SelectBox"
import Option from "@smart-react-components/ui/Select/Option"
import OptionGroup from "@smart-react-components/ui/Select/OptionGroup"
import React from 'react'

export const Palette = () => {
  const [active, setActive] = React.useState(() => null)

  return (
    <>
      <SelectBox active={active} setActive={setActive} placeholder="Label">
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="secondary" active={active} setActive={setActive}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="success" active={active} setActive={setActive}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="danger" active={active} setActive={setActive}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="warning" active={active} setActive={setActive}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="info" active={active} setActive={setActive}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="dynamic" active={active} setActive={setActive}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="!dynamic" active={active} setActive={setActive}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
    </>
  )
}

export const Soft = () => {
  const [active, setActive] = React.useState(() => [])

  return (
    <>
      <SelectBox active={active} setActive={setActive} isSoft>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="secondary" active={active} setActive={setActive} isSoft>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="success" active={active} setActive={setActive} isSoft>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="danger" active={active} setActive={setActive} isSoft>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="warning" active={active} setActive={setActive} isSoft>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="info" active={active} setActive={setActive} isSoft>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="dynamic" active={active} setActive={setActive} isSoft>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="!dynamic" active={active} setActive={setActive} isSoft>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
    </>
  )
}

export const Outline = () => {
  const [active, setActive] = React.useState(() => [])

  return (
    <>
      <SelectBox active={active} setActive={setActive} isOutline={false}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="secondary" active={active} setActive={setActive} isOutline={false}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="success" active={active} setActive={setActive} isOutline={false}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="danger" active={active} setActive={setActive} isOutline={false}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="warning" active={active} setActive={setActive} isOutline={false}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="info" active={active} setActive={setActive} isOutline={false}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="dynamic" active={active} setActive={setActive} isOutline={false}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="!dynamic" active={active} setActive={setActive} isOutline={false}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
    </>
  )
}

export const SoftOutline = () => {
  const [active, setActive] = React.useState(() => [])

  return (
    <>
      <SelectBox active={active} setActive={setActive} isSoft isOutline={false}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="secondary" active={active} setActive={setActive} isSoft isOutline={false}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="success" active={active} setActive={setActive} isSoft isOutline={false}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="danger" active={active} setActive={setActive} isSoft isOutline={false}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="warning" active={active} setActive={setActive} isSoft isOutline={false}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="info" active={active} setActive={setActive} isSoft isOutline={false}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="dynamic" active={active} setActive={setActive} isSoft isOutline={false}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox palette="!dynamic" active={active} setActive={setActive} isSoft isOutline={false}>
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
    </>
  )
}

export const Size = () => {
  const [active, setActive] = React.useState(() => [])

  return (
    <>
      <SelectBox active={active} setActive={setActive} size="small" label="Small">
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox active={active} setActive={setActive} label="Medium">
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox active={active} setActive={setActive} size="large" label="Large">
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
    </>
  )
}

export const Shape = () => {
  const [active, setActive] = React.useState(() => [])

  return (
    <>
      <SelectBox active={active} setActive={setActive} label="Rectangle">
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
      <SelectBox active={active} setActive={setActive} label="Rounded" shape="rounded">
        <OptionGroup label="Category">
          <Option value="item-1">Item 1</Option>
          <Option value="item-2">Item 2</Option>
          <Option value="item-3">Item 3</Option>
        </OptionGroup>
      </SelectBox>
    </>
  )
}
