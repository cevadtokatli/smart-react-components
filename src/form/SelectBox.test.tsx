import React from "react"
import renderTheme from "../test/renderTheme"
import mountTheme from "../test/mountTheme"
import Option from "./Option"
import SelectBox from "./SelectBox"
import InputLabel from "./InputLabel"
import InputError from "./InputError"
import InputAddon from "./InputAddon"

describe("<SelectBox />", () => {
    let children
    let props

    beforeAll(() => {
        children = [
            <Option key="0" value="value-1">Value-1</Option>,
            <Option key="1" value="value-2">Value-2</Option>,
            <Option key="2" value="value-3">Value-3</Option>   
        ]
    })

    beforeEach(() => {
        props = {
            active: "value-2",
            setActive: jest.fn(),
            selectProps: {
                id: "selectbox",
                onClick: e => e.target.click()
            },
            showAnimation: false,
            hideAnimation: false
        }
    })

    it("should match snapshot", () => {
        const wrapper = renderTheme(<SelectBox {...props} label="Label" selectProps={{placeholder:"Placeholder..."}}>{children}</SelectBox>)
        expect(wrapper).toMatchSnapshot()
    })
    
    it("should call setActive", () => {
        const wrapper = mountTheme(<SelectBox {...props}>{children}</SelectBox>)
        wrapper.find("#selectbox").hostNodes().simulate("click")
        wrapper.find(Option).at(0).simulate("click")
        expect(props.setActive).toHaveBeenCalled()
    })

    it("should not call setActive", () => {
        const wrapper = mountTheme(<SelectBox {...props} selectProps={{...props.selectProps,disabled:true}}>{children}</SelectBox>)
        wrapper.find("#selectbox").hostNodes().simulate("click")
        wrapper.find(Option).at(0).simulate("click")
        expect(props.setActive).not.toHaveBeenCalled()
    })

    it("should render label", () => {
        props.setActive = value => props.active = value
        const wrapper = mountTheme(<SelectBox {...props} label="Label">{children}</SelectBox>)
        expect(wrapper.find(InputLabel).exists()).toBe(true)
    })

    it("should render error", () => {
        const wrapper = mountTheme(<SelectBox {...props} error="Error">{children}</SelectBox>)
        expect(wrapper.find(InputError).exists()).toBe(true)
    })

    it("should render left addon", () => {
        const wrapper = mountTheme(<SelectBox {...props} leftAddon="Left Addon">{children}</SelectBox>)
        expect(wrapper.find(InputAddon).exists()).toBe(true)
    })

    it("should render right addon", () => {
        const wrapper = mountTheme(<SelectBox {...props} rightAddon="Right Addon">{children}</SelectBox>)
        expect(wrapper.find(InputAddon).exists()).toBe(true)
    })

    it("should render search input", () => {
        const wrapper = mountTheme(<SelectBox {...props} search={true} searchInputProps={{id:"search-input"}}>{children}</SelectBox>)
        wrapper.find("#selectbox").hostNodes().simulate("click")
        expect(wrapper.find("#search-input").exists()).toBe(true)
    })
})