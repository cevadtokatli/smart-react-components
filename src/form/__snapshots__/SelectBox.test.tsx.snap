// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectBox /> should match snapshot 1`] = `
<div
  class="sc-AxjAm sc-fzokOt gMZWcK"
>
  <label
    class="sc-fzqBZW sc-fzqNJr hvPyVx"
  >
    Label
  </label>
  <div
    class="sc-fzoyAV sc-fzoLag iNbMxX"
    data-src-active="false"
    data-src-placeholder-style="false"
    display="block"
    font-family="inherit"
    placeholder="Placeholder..."
    shape="default"
    tabindex="0"
    width="100%"
  >
    <div
      class="sc-AxjAm bSfLnR"
      display="flex"
    >
      <div
        class="sc-AxjAm jfvPlI"
      >
        Value-2
      </div>
      <svg
        class="sc-AxhUy sc-fzozJi iXmFV"
        viewBox="5 4 14 14"
      >
        <path
          d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
        />
      </svg>
    </div>
  </div>
</div>
`;
